### PR TITLE
Add tiled layout builtin

### PR DESCRIPTION
**Pull request by WojciechBarczynski**
Adds tile layout, that maximizes space covered by input textures on output texture

### CODE REVIEW

**Comment by WojciechBarczynski** [comment]
Review suggestions in the follow up PR

File: compositor_common/src/error.rs [hunk 1]
```diff
@@ -64,6 +64,12 @@ pub enum BuiltinSpecValidationError {
     FixedLayoutLeftRightOnlyOne,
     #[error("Nodes that use transformation \"transform_to_resolution\" need to have exactly one input pad.")]
     TransformToResolutionExactlyOneInput,
+    #[error(
+        "Nodes that use transformation \"tiled_layout\" need to have at most {TILED_LAYOUT_MAX_INPUTS_COUNT}."
```
Comment (wkozyra95) [inline compositor_common/src/error.rs:68]: "Transformation \"tiled_layout\" supports up to {TILED_LAYOUT_MAX_INPUTS_COUNT} inputs."

File: compositor_common/src/scene/builtin_transformations.rs [hunk 2]
```diff
@@ -22,6 +24,13 @@ pub enum BuiltinSpec {
         #[serde(default)]
         background_color_rgba: RGBAColor,
     },
+    TiledLayout {
+        #[serde(default)]
+        background_color_rgba: RGBAColor,
```
Comment (wkozyra95) [inline compositor_common/src/scene/builtin_transformations.rs:29]: Is default fully transparent?

Comment (WojciechBarczynski) [reply compositor_common/src/scene/builtin_transformations.rs:29]: Yep

Comment (WojciechBarczynski) [reply compositor_common/src/scene/builtin_transformations.rs:29]: I planned to add an option to set the background as a node, allowing to set anything that node can produce as a background, but I'm not sure whether mixing it in the next transformation (sth like blend), wouldn't be better. What do you think?

Comment (wkozyra95) [reply compositor_common/src/scene/builtin_transformations.rs:29]: Yes, this should be the responsibility of a different component, either using fixed position layout or we could make an Overlay component with a dynamic resolution that takes 2 inputs  where (resolution is based on one of them)

**Pull request merged**